src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -66,56 +73,72 @@ withMinerResources
     -> IO a
 withMinerResources logger (EnableConfig enabled conf) nid cutDb inner
     | not enabled = inner Nothing
-    | otherwise = inner . Just $ MinerResources
-        { _minerResLogger = logger
-        , _minerResNodeId = nid
-        , _minerResCutDb = cutDb
-        , _minerResConfig = conf
-        }
+    | otherwise = do
+        tms <- newTVarIO Nothing
+        inner . Just $ MinerResources
+            { _minerResLogger = logger
+            , _minerResNodeId = nid
+            , _minerResCutDb = cutDb
+            , _minerResConfig = conf
+            , _minerResState = tms
```

Comment:
Similar to `CutDb`, a new type `MiningState` is now shared between the `working` and `publishing` functions in the `Coordinator` module.